src/ml_tooling/transformers/pandas_transformers.py

```diff
@@ -34,18 +34,38 @@ def transform(self, X):
 # noinspection PyUnusedLocal
 class FillNA(BaseEstimator, TransformerMixin):
     """
-    Fills NA values with given value
+    Fills NA values with given value or strategy. If no value or strategy are
+    supplied missings are imputed with zero. If both a value and a strategy
+    are supplied the strategy will be used.
     """
 
-    def __init__(self, value):
+    def most_freq(X):
+        return pd.DataFrame.mode(X).iloc[0]
+
+    func_map_ = {'mean': pd.DataFrame.mean,
+                 'median': pd.DataFrame.median,
+                 'most_freq': most_freq,
+                 'max': pd.DataFrame.max,
+                 'min': pd.DataFrame.min}
+
+    def __init__(self, value=0, strategy=None):
         self.value = value
+        self.strategy = strategy
+        self.column_values_ = None
 
-    def fit(self, X, y=None):
+    def fit(self, X: pd.DataFrame, y=None):
+        if self.strategy is not None:
+            func = FillNA.func_map_[self.strategy]
```

Comment:
Inside the class, always refer to properties using self
```suggestion
            func = self.func_map_[self.strategy]
```